spec/ot-date.spec.js

```diff
@@ -0,0 +1,99 @@
+"use strict";
+
+describe("OtDate", () => {
+    var OtDate, MomentFake;
+
+    beforeEach(() => {
+        class MomentFake {
+            constructor() {
+                [
+                    "add",
+                    "fromString",
+                    "toDate",
+                    "utc",
+                    "unix"
+                ].forEach((method) => {
+                    this[method] = jasmine.createSpy(method);
+                    this[method].andCallFake(() => {
+                        return this;
+                    });
+                });
+                this.isBefore = jasmine.createSpy("isBefore");
+                this.isBefore.andCallFake((date) => {
+                    return true;
+                });
+                this.toString = jasmine.createSpy("toString");
+                this.toString.andCallFake(() => {
+                    return "a formatted date";
+                });
```

Comment:
Consider having the returned value as part of the list ...  Here is just a partial example

```
methods = {
    add: this,
    fromString: this,
    toDate: this,
    isBefore: true,
    toString: "a formatted date"
};
Object.keys(methods).forEach((method) => {
    this[method] = jasmine.createSpy(method).andReturn(methods[method]);
});